pages/glossary/implicit-section-of-content.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Implicit section of content
+key: implicit-section-of-content
+unambiguous: true
+objective: true
+oudatedLinksIgnore:
+  # ignore below links from `link-is-outdated` test
+  - 'https://www.w3.org/TR/html52/dom.html#heading-content'
+  - 'https://www.w3.org/TR/html52/common-idioms-without-dedicated-elements.html#common-idioms-without-dedicated-elements'
+---
+
+An _implicit section of content_ is all the nodes of the [flat tree][], in tree order, between an initial element with a [semantic role][] of [`heading`][heading] and the next element (in tree order) which has both a [semantic role][] of [`heading`][heading] and an [`aria-level`][aria-level] property lower or equal to that of the initial [`heading`][heading] element. The initial element is included, not the final one.
```

Comment:
I feel like I've left this comment a few times before. In my opinion "the next heading" is not sufficient to describe where the next section starts. The semantics aren't leading in deciding what something "is", the design of a page does. Semantics should represent the visual information, not the other way around. In WCAG we don't fail designs because it doesn't match the semantics, you fail the semantics because it doesn't match the design.

If everyone else thinks i'm wrong about this, feel free to dismiss my comment and move on.

Reply:
I think that you are perfectly right in saying that "visual" sections do exist and should be taken into account. **BUT** I feel this would be topic for another PR, or maybe even another rule for the following reasons:

1. I think that having elements with a role of `heading` which do not describe their implicit section of content **is** a WCAG failure. Maybe not for sighted users, but for users of AT who can list headings and jump to the interesting one, thus need them to describe the content. Therefore, I think the rule as it is now is useful and detect an actual accessibility issue. Maybe not perfect, but useful.
2. The rule has always been targeting elements with a role of `heading`. Thus, it has always ignored visual sections and visual headings. It has also always targeted elements that are in the accessibility tree, but not visible (Passed example 6) and thus never was concerned by visual headings and sections. This doesn't mean we can't or shouldn't change it, but it means that considering visual section is more than just fixing the current definition.
3. The fact that this PR essentially doesn't change the rule that use the definition is a strong indication for me that it is an atomic PR making the minimum change and that changing the rule should be topic of another PR.

Now, and in relation to the first point, I do agree that having visual headings that don't describe their visual section of content is also a WCAG failure. And as you point out, we can't assume that the programmatic and visual structures are identical. Therefore, it would be nice to also handle visual sections. To me it is however a different topic than the current one. I do think it should be a completely separate rule because both kinds of section need a descriptive heading, and trying to mix them lead to a very complex rule. I kept the name "**implicit** section of content" to leave room (to us) for more kind of sections of content, and to indicate (to readers) that there could be other kinds of sections of content.

**TLDR:** Descriptive headings for visual sections of content are needed, but I think as a separate rule.

Reply:
This has been going on for months, and I don't think there is anything I can say that I haven't already covered. Here's the basic argument. Based on this, lets have a vote in this week's CG call so we can resolve this.


1. There aren't different "types" of headings or sections as far as WCAG is concerned. There are headings, and there are sections and they can either have semantic markup, or they can not.
2. SC 1.3.1 requires that all headings be marked up as headings. It does not require that sections are marked up as sections. Therefore assumptions based on headings being marked up correctly are more acceptable than those that assume sections are.
3. This rule relies on the assumption that sections are marked up "implicitly", isn't required by WCAG, and isn't the only way sections can be marked up.

Therefore, in my opinion this is not an acceptable assumption for this rule.

Please :+1: if you agree, and :-1: if you disagree with this.

Reply:
> 1. There aren't different "types" of headings or sections as far as WCAG is concerned. There are headings, and there are sections and they can either have semantic markup, or they can not.

"Headings that have semantic markup" and "heading that don't have semantic markup" are, _de facto_, two "types" of headings. I agree that WCAG treat them the same way, but it doesn't mean we have to. This rule is, and has always been, a partial check for WCAG that is only considering "heading that have semantic markup" (and has always only considered those).

> 2. SC 1.3.1 requires that all headings be marked up as headings. It does not require that sections are marked up as sections. Therefore assumptions based on headings being marked up correctly are more acceptable than those that assume sections are.

I don't fully agree. I think that sections are part of "structure" that is required to match by 1.3.1. Furthermore, 2.4.10 does require sections to start with headings (in the generic sense of the term). Granted, it's only AAA.

> 3. This rule relies on the assumption that sections are marked up "implicitly", isn't required by WCAG, and isn't the only way sections can be marked up.

I disagree. I think that cases that fail this rule do fail 2.4.6. True, it is possible to pass this rule and nonetheless fail 2.4.6, but that is a very frequent case for our rules and does not need an assumption.

If I take your previous codepen: https://codepen.io/wilcofiers/pen/abOvWRB where the "sections defined by visual cues" and "section computed from `role="heading"` structure" don't match, I think that it fails 2.4.6. Because this is what Chrome+NVDA reads (newlines added by me):
> Hello world  heading    level 1  
Welcome to Lorem  heading    level 2  
Lorem ipsum, dolor sit amet consectetur adipisicing elit. Tenetur excepturi nobis illum ad officia hic, nesciunt distinctio quos iusto saepe dicta, incidunt modi vero et enim veritatis esse! Quae, fuga?  
Welcome to Ipsum  heading    level 2  
Lorem ipsum, dolor sit amet consectetur adipisicing elit. Tenetur excepturi nobis illum ad officia hic, nesciunt distinctio quos iusto saepe dicta, incidunt modi vero et enim veritatis esse! Quae, fuga?  Lorem ipsum dolor sit amet consectetur adipisicing elit. Molestiae porro voluptates repellat ducimus necessitatibus blanditiis quia ipsa? Quis libero hic voluptates aliquid, corporis quidem asperiores cumque, a officiis cum possimus.

Screen reader users have no clue that the "Welcome to Ipsum" section ends early. For them, the last bit is all part of the section that started at that heading.


Again, I am not saying that "visual sections" don't exists or that we shouldn't consider them. I am saying that considering them is another matter, topic for another PR or likely even another rule; and that failing this rule is indeed failing 2.4.6.